test/integration/image-optimizer/test/disable-write-to-cache-dir.test.ts

```diff
@@ -0,0 +1,13 @@
+import { join } from 'path'
+import { setupTests } from './util'
+
+const appDir = join(__dirname, '../app')
+const imagesDir = join(appDir, '.next', 'cache', 'images')
+
+describe('with disableWriteToCacheDir config', () => {
+  setupTests({
+    nextConfigImages: { disableWriteToCacheDir: true },
+    appDir,
+    imagesDir,
+  })
+})
```

Comment:
In order to test this, you'll need to check the contents of `imagesDir` to make sure its empty.

So all the tests that call `fsToJson()` will need to look for the config flag and assert the filesystem is empty.

Like this test for example:

https://github.com/vercel/next.js/blob/9435b7b9602136aeac77f5ff55bb553dcaa2b01a/test/integration/image-optimizer/test/util.ts#L876